<launch>
    <!-- Subscribed topics -->
    <param name="LiDAR_pointcloud_topic" type="string" value= "/laser_cloud_flat" />
    <param name="IMU_topic" type="string" value= "/imu/imu" />
    <param name="Image_topic" type="string" value= "/left/image_raw" />
    <param name="lvi_fusion_common/map_output_dir" type="string" value="$(env HOME)/lvi_fusion_output" />
    <rosparam command="load" file="$(find lvi_fusion)/../config/lvi_fusion_config_ntu.yaml" />

    <node pkg="lvi_fusion" type="lvi_fusion_LiDAR_front_end" name="lvi_fusion_LiDAR_front_end"  output="screen" required="true"/>
    <node pkg="lvi_fusion" type="lvi_fusion_mapping" name="lvi_fusion_mapping" output="screen" required="true"/>
    
    <arg name="rviz" default="1" />
    <group if="$(arg rviz)">
        <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find lvi_fusion)/../config/rviz/lvi_fusion_rviz_config.rviz" />
    </group>
</launch>
